pkg/controller/cheinstallation/cheinstallation_controller_test.go

```diff
@@ -105,36 +97,6 @@ func TestCheInstallationController(t *testing.T) {
 			AssertThatSubscription(t, cheSub.Namespace, cheSub.Name, cl).
 				Exists().
 				HasSpec(cheSub.Spec)
-
-			AssertThatSubscription(t, tektonSub.Namespace, tektonSub.Name, cl).
-				DoesNotExist()
-
-		})
-
-		t.Run("should create tekton subscription and requeue", func(t *testing.T) {
-			// when
-			result, err := r.Reconcile(request)
-
-			// then
-			require.NoError(t, err)
-
-			assert.True(t, result.Requeue)
-			AssertThatNamespace(t, cheOperatorNs, cl).
-				Exists().
-				HasLabels(toolchain.Labels())
-
-			AssertThatOperatorGroup(t, cheOg.Namespace, cheOg.Name, cl).
-				Exists().
-				HasSize(1).
-				HasSpec(cheOg.Spec)
-
-			AssertThatSubscription(t, cheSub.Namespace, cheSub.Name, cl).
-				Exists().
-				HasSpec(cheSub.Spec)
-
-			AssertThatSubscription(t, tektonSub.Namespace, tektonSub.Name, cl).
-				Exists().
-				HasSpec(tektonSub.Spec)
 		})
 
 		t.Run("should not requeue", func(t *testing.T) {
```

Comment:
sure, that's true, But I believe, it's good to have this kind of tests to make sure that everything works correctly, instead of depending on e2e tests to debug/fix any issue

Reply:
I didn't want to say that we should depend only on e2e tests, the unit tests are needed for sure, but this is not really a unit test. Instead of depending on the previous subtests it should contain a proper test setup so the test is isolated and can be executed on its own. 
In case you don't want to do the proper setup for every subtest separately, then another option would be embedding the subtests into the one it depends on - in other words, the `"should not requeue"` would be a subtest of `"should create che subscription and requeue"` and that would be a subtest of `"should create operator group and requeue"` and so on - to make it clear that they depend on the previous ones

Reply:
thanks for explaination. Let's see how can we fix it in Che refactoring PR